lib/image_processing/vips.rb

```diff
@@ -0,0 +1,231 @@
+require "vips"
+require_relative "vips/color"
+require_relative "vips/gravity"
+require "tempfile"
+
+if Gem::Version.new(Vips::VERSION) < Gem::Version.new("2.0.0")
+  raise "image_processing requires ruby-vips version >= 2.0.0"
+end
+
+module ImageProcessing
+  module Vips
+    module_function
+
+    # Changes the image encoding format to the given format
+    # Libvips convert the file automatically based on the name
+    # since we want the user to have a destructive and a non destructive method
+    # using the destructive method will cause the original file to be deleted
+    #
+    # @param [File, Tempfile] file     the image to convert
+    # @param [String] format           the format to convert to
+    # @param [File] destination        the end destination where the image will be safe
+    # @return [File, Tempfile]
+    def convert(file, format, destination: nil, &block)
+      vips_image = ::Vips::Image.new_from_file file.path
+      destination_image = destination || _tempfile(".#{format}")
+      vips_image.write_to_file(destination_image.path)
+      destination_image
```

Comment:
Is it possible to reuse `with_ruby_vips` here?

Reply:
I think we can try.
I quick sketch would look like 
```ruby
def convert(file, format, destination: nil, &block)
  with_ruby_vips(file, destination || _tempfile(".#{format}"))
end

def convert!(file, format, &block)
    destination_path = Pathname(file.path).sub_ext(".#{format}").to_s
    with_ruby_vips(file, File.new(destination_path)).tap do
      file.close
     File.delete(file.path)
  end
end
```

Reply:
Yes, that looks about right, though it would probably make sense for `#convert!` to reuse `#convert` instead of also calling `#with_ruby_vips`.

Note that if the destination will be a string path (https://github.com/janko-m/image_processing/pull/14#discussion_r143354447), I think it would be simplest to add an `:extension` option to `#with_ruby_vips`, and then do this:

```rb
def convert(file, format, destination: nil, &block)
  with_ruby_vips(file, destination, extension: ".#{format}")
end
```

In `#with_ruby_vips`, if `destination` is given, then `:extension` can be ignored, otherwise it's used when creating a tempfile.

Reply:
Yes I was thinking of something along this lines. Yesterday simply I was to tire. I will continue working today on it.